### PR TITLE
Optionally use globs to specify the source files

### PR DESCRIPTION
**Pull request by kring**
Rather than regular expressions hardcoded into the source.

### CODE REVIEW

**Comment by kring** [comment]
This is to help with TerriaJS/terriajs#1944.

File: generateSchema.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ var esprima = require('esprima'),
     path = require('path'),
     when = require('when'),
     node = require('when/node'),
-    fsp = node.liftAll(require('fs')); // promisified version of fs
+    fsp = node.liftAll(require('fs')),
+    glob = require('glob-all'); // promisified version of fs
```
Comment (stevage) [inline generateSchema.js:11]: this comment got displaced.

File: generateSchema.js [hunk 2]
```diff
@@ -437,14 +439,25 @@ function processModels() {
         return model && model.typeId;
     }
     function isSchemable(modelName) {
-        return modelName.match(/Catalog(Item|Group|Member)\.js$/) &&                   
+        return modelName.match(/Catalog(Item|Group|Member)\.js$/) &&
               !modelName.match(/(ArcGisMapServerCatalogGroup|addUserCatalogMember)/);
```
Comment (stevage) [inline generateSchema.js:442]: I don't really remember why ArcGisMapServerCatalogGroup is a special case, but perhaps that should be generalised too?

Comment (kring) [reply generateSchema.js:442]: It was deprecated at one point, kind of like `AsyncFunctionResultCatalogItem` is now.  But Arthur resurrected it recently, so it should no longer be excluded.

File: generateSchema.js (hunk 1 above)
Comment (kring) [reply generateSchema.js:11]: Thanks, fixed it in master.

**Pull request merged**